helpers/mysql/lib/opentelemetry/helpers/mysql.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0module OpenTelemetry
+
+module OpenTelemetry
+  module Helpers
+    # This class contains helpers for MySQL instrumentation libraries, like mysql2 and
+    # trilogy. It is intended for use by instrumentation developers and not to
+    # be called directly by an application.
+    #
+    # To use this in your instrumentation, the `Instrumentation` class for
+    # your gem must contain configuration options for:
+    #  * `:span_name`
+    #  Example:
+    #  `option :span_name, default: :statement_type, validate: %I[statement_type db_name db_operation_and_name]`
+    #
+    # @api public
+    class MySQL
+      QUERY_NAMES = [
+        'set names',
+        'select',
+        'insert',
+        'update',
+        'delete',
+        'begin',
+        'commit',
+        'rollback',
+        'savepoint',
+        'release savepoint',
+        'explain',
+        'drop database',
+        'drop table',
+        'create database',
+        'create table'
+      ].freeze
+
+      QUERY_NAME_REGEX = Regexp.new("^(#{QUERY_NAMES.join('|')})", Regexp::IGNORECASE)
+      EXTRACT_STATEMENT_FAILURE_MESSAGE = 'Error extracting SQL statement type'
+
+      # This is a span naming utility intended for use in MySQL database
+      # adapter instrumentation.
+      #
+      # @param sql [String] The SQL statement for the span.
+      # @param operation [String] The database operation.
+      # @param database_name [String] The name of the database.
+      # @param config [Hash] The user's configuration for the database adapter.
+      #  Desired keys:
+      #    * `:span_name` => A symbol describing the type of name desired. Expected options are `:statement_type`, `:db_name`, and `:db_operation_and_name`. A nil or unknown `:span_name` will return 'mysql' as the span name
+      # @return [String] The span name.
+      # @api public
+      def self.database_span_name(sql, operation, database_name, config)
+        case config[:span_name]
+        when :statement_type
+          extract_statement_type(sql)
+        when :db_name
+          database_name
+        when :db_operation_and_name
+          db_operation_and_name(operation, database_name)
+        end || 'mysql'
+      end
+
+      # @api private
+      def self.extract_statement_type(sql)
+        QUERY_NAME_REGEX.match(sql) { |match| match[1].downcase } unless sql.nil?
+      rescue StandardError => e
+        # TODO: Replace with `handle_error` Issue #609
+        OpenTelemetry.logger.error("#{EXTRACT_STATEMENT_FAILURE_MESSAGE}: #{e.message}")
```

Comment:
There's not really any value in extracting the prefix here to a constant. I'd rather just have the text directly here (it isn't referenced anywhere else as far as I can tell).

Can we address the `TODO` now or is there a larger issue blocking it?

Reply:
Thanks for the feedback, @fbogsany! I removed the constant and took care of the work for #609 in the MySQL and SQL Obfuscation gems. 

See: f69eb9ce478fa5627b4c7bea9d07332cc4b00f6b 

Initially, I wanted to put that work in a separate PR to make the first version of the new helper gems identical in behavior to the code extracted out of the instrumentation gems. However, with a closer look, I realized there were already other behavior changes. 

I'm happy to back out the changes and put them in a separate PR if you'd prefer that approach.

Reply:
There was a little more work needed here to pass the test cases in mysql2 and trilogy: f50f16b6358cfd0843ca9cb9f1d691ffb78dd324

Reply:
❤️ Thank you! I'll review again today.